profile.d/toolbox.sh

```diff
@@ -11,9 +11,7 @@ toolbox_welcome_stub="$toolbox_config/toolbox-welcome-shown"
 # shellcheck disable=2046
 # shellcheck disable=SC1091
 eval $(
-          if [ -f /etc/os-release ]; then
-              . /etc/os-release
-          else
+          if ! . /etc/os-release 2>/dev/null; then
```

Comment:
As per the POSIX [standard](https://pubs.opengroup.org/onlinepubs/9699919799.2013edition/utilities/V3_chap02.html#dot), a non-interactive POSIX shell is supposed to exit straight away if the file being sourced is not found.

This can be verified with this script:
```
#!/bin/sh

if ! . /etc/os-release; then
    echo "falling back"
fi
```

... or this:
```
#!/bin/sh

. /etc/os-release || echo "falling back"
```

A shell like Bash will behave differently.  It won't exit straight away, and we can handle the failure as expected.

However `profile.d/toolbox.sh` is sourced by various different shells (eg., Z shell), so we need to be as [pedantic](https://github.com/containers/toolbox/commit/88f2916822f1fdf71caff9d376587adc86b57f20) as possible.  See https://github.com/containers/toolbox/pull/660 for an example of a Z shell incompatibility.